### PR TITLE
systemtest: migrate TLS cert tests to Go

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Migrate TLS system tests to Go. Only essentials are migrated, some less important configurations (e.g. optional client certs) are no longer tested.

We now capture net/http server error logs with logp. Previously they were sent to the standard library's log package, which logp disables by default.

See commit message for program used to generate client certificate/key pair.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
~- [ ] Documentation has been updated~

## How to test these changes

1. Run apm-server with `-E apm-server.ssl.client_authentication=required`
2. `curl -k https://localhost:8200`; should fail because no client cert is provided
3. Check apm-server logs for an error message with logger "beater.http" indicating the client did not provide a certificate

## Related issues

None

### CODE REVIEW

File: beater/http.go [hunk 1]
```diff
@@ -60,6 +63,7 @@ func newHTTPServer(
 		ReadTimeout:    cfg.ReadTimeout,
 		WriteTimeout:   cfg.WriteTimeout,
 		MaxHeaderBytes: cfg.MaxHeaderSize,
+		ErrorLog:       newErrorLog(logger),
```
Comment (stuartnelson3) [inline beater/http.go:66]: cool, TIL

**Review by stuartnelson3 — APPROVED** [review]

**Comment by simitt** [comment]
🎉

**Comment by marclop** [comment]
Tested locally with `7.16.0 BC2`:

1. `./apm-server -E apm-server.ssl.client_authentication=required -E output.elasticsearch.username=admin -E output.elasticsearch.password=changeme -e`
2. Verified that `certificate file not configured accessing 'apm-server.ssl'` is printed in the logs:
```json
{"log.level":"info","@timestamp":"2021-11-04T10:40:10.246+0800","log.origin":{"file.name":"instance/beat.go","file.line":683},"message":"Home path: [/Users/marclop/Downloads/apm-server-7.16.0-darwin-x86_64] Config path: [/Users/marclop/Downloads/apm-server-7.16.0-darwin-x86_64] Data path: [/Users/marclop/Downloads/apm-server-7.16.0-darwin-x86_64/data] Logs path: [/Users/marclop/Downloads/apm-server-7.16.0-darwin-x86_64/logs]","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-04T10:40:10.285+0800","log.origin":{"file.name":"instance/beat.go","file.line":691},"message":"Beat ID: a61ddcd7-ff3b-43ac-8585-857c8c231eff","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-04T10:40:10.287+0800","log.logger":"beat","log.origin":{"file.name":"instance/beat.go","file.line":1037},"message":"Beat info","service.name":"apm-server","system_info":{"beat":{"path":{"config":"/Users/marclop/Downloads/apm-server-7.16.0-darwin-x86_64","data":"/Users/marclop/Downloads/apm-server-7.16.0-darwin-x86_64/data","home":"/Users/marclop/Downloads/apm-server-7.16.0-darwin-x86_64","logs":"/Users/marclop/Downloads/apm-server-7.16.0-darwin-x86_64/logs"},"type":"apm-server","uuid":"a61ddcd7-ff3b-43ac-8585-857c8c231eff"},"ecs.version":"1.6.0"}}
{"log.level":"info","@timestamp":"2021-11-04T10:40:10.287+0800","log.logger":"beat","log.origin":{"file.name":"instance/beat.go","file.line":1046},"message":"Build info","service.name":"apm-server","system_info":{"build":{"commit":"73339c29675f6f987cd96ffac8d4b24284f0275b","libbeat":"7.16.0","time":"2021-10-27T20:19:32.000Z","version":"7.16.0"},"ecs.version":"1.6.0"}}
{"log.level":"info","@timestamp":"2021-11-04T10:40:10.287+0800","log.logger":"beat","log.origin":{"file.name":"instance/beat.go","file.line":1049},"message":"Go runtime info","service.name":"apm-server","system_info":{"go":{"os":"darwin","arch":"amd64","max_procs":8,"version":"go1.17.1"},"ecs.version":"1.6.0"}}
{"log.level":"info","@timestamp":"2021-11-04T10:40:10.287+0800","log.logger":"beat","log.origin":{"file.name":"instance/beat.go","file.line":1053},"message":"Host info","service.name":"apm-server","system_info":{"host":{"architecture":"x86_64","boot_time":"2021-10-29T23:21:08.323819+08:00","name":"Marcs-MacBook-Pro-2.local","ip":["127.0.0.1/8","::1/128","fe80::1/64","fe80::1c00:39ff:fe02:7c65/64","fe80::1c00:39ff:fe02:7c64/64","fe80::1ce4:ddef:c2d9:c128/64","192.168.1.220/24","fe80::840b:b1ff:fe68:9f3f/64","fe80::840b:b1ff:fe68:9f3f/64","fe80::b8e9:33bf:14a0:b930/64","fe80::1ece:c781:5352:75d6/64","10.211.55.2/24","fe80::1079:666f:b042:e1b7/64","10.37.129.2/24","fe80::61:ada0:3f8c:4e72/64"],"kernel_version":"20.6.0","mac":["1e:00:39:02:7c:65","1e:00:39:02:7c:64","1e:00:39:02:7c:44","1e:00:39:02:7c:45","36:88:28:18:df:40","36:88:28:18:df:44","36:88:28:18:df:40","3e:06:30:06:7e:d3","3c:06:30:06:7e:d3","86:0b:b1:68:9f:3f","86:0b:b1:68:9f:3f","86:cf:e9:60:e8:63","3e:06:30:60:34:64","0a:50:61:6d:13:1d","3e:06:30:60:34:65"],"os":{"type":"macos","family":"darwin","platform":"darwin","name":"Mac OS X","version":"10.16","major":10,"minor":16,"patch":0,"build":"20G165"},"timezone":"HKT","timezone_offset_sec":28800,"id":"3173F67B-7E78-50BC-925B-742DFF799ACD"},"ecs.version":"1.6.0"}}
{"log.level":"info","@timestamp":"2021-11-04T10:40:10.287+0800","log.logger":"beat","log.origin":{"file.name":"instance/beat.go","file.line":1082},"message":"Process info","service.name":"apm-server","system_info":{"process":{"cwd":"/Users/marclop/Downloads/apm-server-7.16.0-darwin-x86_64","exe":"./apm-server","name":"apm-server","pid":9845,"ppid":7745,"start_time":"2021-11-04T10:40:06.038+0800"},"ecs.version":"1.6.0"}}
{"log.level":"info","@timestamp":"2021-11-04T10:40:10.287+0800","log.origin":{"file.name":"instance/beat.go","file.line":324},"message":"Setup Beat: apm-server; Version: 7.16.0","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-04T10:40:10.289+0800","log.logger":"esclientleg","log.origin":{"file.name":"eslegclient/connection.go","file.line":102},"message":"elasticsearch url: http://localhost:9200","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-04T10:40:10.290+0800","log.logger":"publisher","log.origin":{"file.name":"pipeline/module.go","file.line":113},"message":"Beat name: Marcs-MacBook-Pro-2.local","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-04T10:40:10.290+0800","log.origin":{"file.name":"instance/beat.go","file.line":458},"message":"apm-server stopped.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"error","@timestamp":"2021-11-04T10:40:10.290+0800","log.origin":{"file.name":"instance/beat.go","file.line":1012},"message":"Exiting: Error processing configuration: certificate file not configured accessing 'apm-server.ssl' (source:'command line flag')","service.name":"apm-server","ecs.version":"1.6.0"}
Exiting: Error processing configuration: certificate file not configured accessing 'apm-server.ssl' (source:'command line flag')
```

It fails to start though, doesn't even start serving http requests, I assume this is the desired behavior but just want to confirm with @axw since it differs from the steps in this PR.

**Comment by axw** [comment]
@marclop ah yeah sorry, I missed a bit: it is also necessary to configure the server cert/key. There's a pair under testdata for testing.

**Comment by marclop** [comment]
Oh I see, I overlooked that, tested again with `7.16.0 BC2`, works as expected:

1. `cp ~/repos/elastic/apm-server/systemtest/apmservertest/{key,cert}.pem ~/Downloads/apm-server-7.16.0-darwin-x86_64/`
2. `./apm-server -E apm-server.ssl.client_authentication=required -E output.elasticsearch.username=admin -E output.elasticsearch.password=changeme -E apm-server.ssl.certificate=cert.pem -E apm-server.ssl.key=key.pem -e`
3. `curl -k https://localhost:8200`
```
curl: (35) error:1401E412:SSL routines:CONNECT_CR_FINISHED:sslv3 alert bad certificate
```
4. Server logs outputs:
```json
{"log.level":"error","@timestamp":"2021-11-04T10:58:25.599+0800","log.logger":"beater.http","log.origin":{"file.name":"http/server.go","file.line":3157},"message":"http: TLS handshake error from 127.0.0.1:62548: tls: client didn't provide a certificate","service.name":"apm-server","ecs.version":"1.6.0"}
```

**Pull request merged**